src/scipp/io/hdf5.py

```diff
@@ -32,6 +32,15 @@ def _as_hdf5_type(a):
     return a
 
 
+def collection_element_name(name, index):
+    """
+    Convert name into an ASCII string that can be used as an object name in HDF5.
+    """
+    ascii_name = name.replace('.', '&#46;').replace('/', '&#47;').encode(
+        'ascii', 'xmlcharrefreplace').decode('ascii')
+    return f'elem_{index:03d}_{ascii_name}'
```

Comment:
Are you including the index here in case we want to preserve order?

Reply:
No. My initial version simply dropped all non-ascii chars. So it was necessary to distinguish names in this way in case all chars are dropped.
But it is not really needed now. Unless we want to switch to ordered dictionaries. There was some talk about that...

Reply:
We may want to have ordered dicts, so maybe keep it? Or not, because it currently implies an order even if there isn't one? But then we would break files later...

Can you completely drop the prefix now and use only the encoded name?

Reply:
> Can you completely drop the prefix now and use only the encoded name?

Yes. It would save >~ 9 bytes per item, which is hardly relevant. I'd rather keep it to avoid the potential for breaking it again in the future.

Reply:
I was not concerned about the size 😄 

It was mainly about having the file "human readable" and as close to the actual structure as possible.

Somewhat related: I recently learned about [HDF5 Dimension Scales](https://docs.h5py.org/en/stable/high/dims.html) and wondered whether we should be using that?

Reply:
Have to come back to this question again... if we drop the prefix, would most old files be compatible with new scipp? And vice versa, if we keep the prefix, it would imply that *all* old files need updating, even if they do not contain special characters?

Reply:
It doesn't affect compatibility. It doesn't parse the group name but look for the `name` attribute. This could be changed but then we would need to write a decoder for XML codes. (Don't think there is anything in the standard library)

Reply:
I don't understand. Below we use
```python
                var_group_name = collection_element_name(name, i)
```
i.e., the group name will now include the prefix?

Reply:
Yes. But the reader does not care. You can add the name attr to old files without changing the group name and everything will work. I added the name change to the migration script to keep things consistent.

Reply:
> But the reader does not care

Are you saying that the reader works on files without the prefix?

Reply:
Yes, As long as the attribute exists

Reply:
Does the attribute exist for legacy files, or was that newly added?

Reply:
It was newly added

Reply:
So you are saying that because we now require the attribute we *anyway* break all old files, so changing group/dataset names is irrelevant?

Reply:
> So you are saying that because we now require the attribute we anyway break all old files

Yes

> so changing group/dataset names is irrelevant?

Essentially, yes. The question is, *if* we want to use a prefix for ordering, we should make all files consistent now. But if we don't care enough about the order, then there is no reason to change the names.

Reply:
As we are moving more and more to become a "Python" library (over the past couple of years) I think it is likely that we will eventually want to preserve order, i.e., keep the prefix. 👍